tools/eslint-rules/prefer-common-mustnotcall.js

```diff
@@ -11,29 +11,18 @@
 const msg = 'Please use common.mustNotCall(msg) instead of ' +
             'common.mustCall(fn, 0) or common.mustCall(0).';
 
-function isCommonMustCall(node) {
-  return node &&
-         node.callee &&
-         node.callee.object &&
-         node.callee.object.name === 'common' &&
-         node.callee.property &&
-         node.callee.property.name === 'mustCall';
-}
-
-function isArgZero(argument) {
-  return argument &&
-         typeof argument.value === 'number' &&
-         argument.value === 0;
-}
-
 module.exports = function(context) {
+  function report(node) {
+    context.report(node, msg);
+  }
+
   return {
-    CallExpression(node) {
-      if (isCommonMustCall(node) &&
-          (isArgZero(node.arguments[0]) ||  //  catch common.mustCall(0)
-           isArgZero(node.arguments[1]))) { //  catch common.mustCall(fn, 0)
-        context.report(node, msg);
-      }
-    }
+    // Catch common.mustCall(0)
+    // eslint-disable-next-line max-len
+    'CallExpression[callee.object.name="common"][callee.property.name="mustCall"][arguments.0.value=0]': report,
```

Comment:
This could be extracted into a function outside of `module.exports` that takes input for the argument position, so something like `getAstSelector(pos)`. Then we also don't need to have the run on line.